openshift/helper/__init__.py

```diff
@@ -128,7 +168,120 @@ def __lookup_method(self, operation, namespaced):
+            },
+            # path to a config file
+            'kubeconfig': {},
+
+            # kubectl context name
+            'context': {}
+        }
+
+        if 'metadata' not in self.properties.keys():
+            raise OpenShiftException(
+                "Object {} does not contain metadata field".format(self.model)
+            )
+
+        if self.namespaced:
+            argument_spec['namespace'] = {
+                'required': True,
+                'property_path': ['metadata', 'name']
+            }
+
+        # mutually_exclusive = None
+        # required_together = None
+        # required_one_of = None
+        # required_if = None
+
+        argument_spec.update(self.__transform_properties(self.properties))
+        return argument_spec
+
+    def __transform_properties(self, properties, prefix='', path=[]):
+        '''
+        Convert a list of properties to an argument_spec dictionary
+
+        :param properties: List of properties from self.properties_from_model_obj()
+        :param prefix: String to prefix to argument names.
+        :return: dict
+        '''
+        args = {}
+        for prop, prop_attributes in properties.items():
+            if prop == 'status':
+                # Do not expose Status to argument spec
+                continue
+            elif prop_attributes['immutable']:
+                # Property cannot be set by the user
+                continue
+            elif prop == 'metadata':
+                # Filter metadata properties added to argument spec
+                if prop_attributes['class'].__name__ != METACLASS_NAME:
+                    raise OpenShiftException("Unknown metadata type: {}".format(prop_attributes['class'].__name__))
+                args['labels'] = {'required': False, 'type': 'dict',
+                                  'property_path': ['metadata', 'labels']}
+                args['annotations'] = {'required': False, 'type': 'dict',
+                                       'property_path': ['metadata', 'labels']}
+            elif prop_attributes['class'].__name__ not in ['int', 'str', 'bool', 'list', 'dict']:
+                # Adds nested properties recursively
+
+                # As we traverse deeper into nested properties, we prefix the final primitive property name with the
+                # chain of property names. For example, 'pod_podsecuritycontext_run_as_user', where 'run_as_user' is
+                # the primitive.
+                #
+                # This may be too hacky, but trying to remove redundant prefixes and generic, non-helpful
+                # prefixes (e.g. Sepc, Template).
+                label = prop_attributes['class'].__name__\
+                        .replace(self.api_version.capitalize(), '')\
+                        .replace(BASE_API_VERSION, '')\
+                        .replace(self.base_model_name, '')\
+                        .replace('Spec', '')\
+                        .replace('Template', '')\
```

Comment:
I'm wondering if we should not strip out Spec and Template. It would definitely make the variable names longer, but I worry that it would cause more confusion. @soltysh any ideas?